src/olympia/reviewers/views.py

```diff
@@ -733,14 +752,41 @@ def review(request, addon, channel=None):
             request, ugettext('Self-reviews are not allowed.'))
         return redirect(reverse('reviewers.dashboard'))
 
+    # Queryset to be paginated for versions. We use the default ordering to get
+    # most recently created first (Note that the template displays each page
+    # in reverse order, older first).
+    versions_qs = (
+        # We want to load all Versions, even deleted ones, while using the
+        # addon.versions related manager to get `addon` property pre-cached on
+        # each version.
+        addon.versions(manager='unfiltered_for_relations')
+             .filter(channel=channel)
+             .select_related('autoapprovalsummary')
+        # Add activity transformer to prefetch all related activity logs on
+        # top of the regular transformers.
+             .transform(Version.transformer_activity)
+    )
+
+    # Specific versions for the reviewer to look at / act upon when POSTing.
+    needs_human_review_versions = []
+    needs_human_review_versions_ids = (
+        request.POST.get(
+            'needs_human_review_versions_ids',
+            request.GET.get('needs_human_review_versions_ids', ''))
```

Comment:
so the versions table will include the versions that need human review too?  If there are a number of versions it'll make the page pretty long with the duplicate version details.  I'm not sure how likely a UX review is on a reviewer tools patch but have Product at least okay'd this layout?

e.g. Don't display a separate list of versions but highlight the specific versions within the main version list?

Reply:
The problem with highlighting the versions in the page is that you then potentially miss a few, and then you don't know which ones to unset the flag on at submission, since they weren't all shown.

My idea was, when this flag is set by scanners, we should react quickly, and don't let the versions needing review accumulate on any given add-on, so the lack of pagination shouldn't be too much of a problem.

Reply:
(Note: I do highlight the versions that have the flag set in the "main" history section, regardless of whether or not the query parameter is present)

Reply:
I don't have an answer either to the problem either - that's why I was pushing chosing the UX solution onto someone in Product 😉

Reply:
@wagnerand has already seen this in slack, but looping him again and @jvillalobos to see if they have suggestions.

Reply:
I agree with @eviljeff that having two lists is a bit much. It's creating duplicate info and making an already very complex page more complex. It's not unlikely that many versions pending human review will accumulate, so that list may become a problem. It's also unlikely that if an add-on accumulates 5+ versions that need human review, that all of them will actually end up being reviewed, instead of maybe the last version or two.

I think that highlighting the versions in the history should be good enough, and it would be the reviewer's responsibility to figure out which versions actually need to be reviewed before submitting their approval.

Reply:
We've been discussing that over IRC with @wagnerand earlier and reached broadly similar conclusions. 

I haven't pushed those changes yet but this is where I landed:
![Screenshot_2019-10-29 WebExt-eval-test Reviewer Tools Add-ons for Firefox(2)](https://user-images.githubusercontent.com/187006/67807266-ca70ba80-fa94-11e9-8ae1-ed30a1eefbd6.png)

Versions needing human review are highlighted in the history, and if there are more needing attention outside of the paginated range there is a mention of that.